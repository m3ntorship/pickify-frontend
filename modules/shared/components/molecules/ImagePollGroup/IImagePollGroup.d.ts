import type { IPostFeed } from '../../../types/postFeed/IPostFeed';

declare namespace IImagePollGroup {
  export interface IProps {
    group: IPostFeed.IGroup;
    addOneVote: (optionId: string, grouId: string) => void;
  }
}

export { IImagePollGroup };
