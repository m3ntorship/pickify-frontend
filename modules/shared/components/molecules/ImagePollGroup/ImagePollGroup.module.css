.image-poll-view {
  @apply sm:grid sm:gap-x-2 sm:gap-y-4;
}
