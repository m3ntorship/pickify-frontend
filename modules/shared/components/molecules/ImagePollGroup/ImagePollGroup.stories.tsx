import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import ImagePollGroup from './ImagePollGroup';
import type { IImagePollGroup } from './IImagePollGroup';

export default {
  title: 'Molecules/ImagePollGroup',
  component: ImagePollGroup,
};

const Template: Story<IImagePollGroup.IProps> = (args): ReactElement => (
  <ImagePollGroup {...args} />
);
export const Default = Template.bind({});
Default.args = {
  group: {
    id: '03644270-7171-4147-b5a1-4233ff547f7a',
    name: 'Group Name',
    options: [
      {
        id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
        body: 'ali tramsy',
        media: [{ url: 'https://placeimg.com/640/480/any' }],
      },
      {
        id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
        body: ' ahmed essam',
        media: [{ url: 'https://placeimg.com/640/480/any' }],
      },
      {
        id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
        body: ' omar gamal',
        media: [{ url: 'https://placeimg.com/640/480/any' }],
      },
    ],
  },
};
