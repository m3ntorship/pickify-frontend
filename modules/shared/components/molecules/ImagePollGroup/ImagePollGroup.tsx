import React from 'react';
import type { FC, ReactElement, MouseEvent } from 'react';
import classNames from 'classnames';
import type { IImagePollGroup } from './IImagePollGroup';
import ImagePollOption from '../ImagePollOption/ImagePollOption';
import styles from './ImagePollGroup.module.css';
import SignleImagePollOption from '../SignleImagePollOption/SignleImagePollOption';
import { getVotesResults } from '../../../logic/votesLogic/votesLogic';

const ImagePollGroup: FC<IImagePollGroup.IProps> = ({
  group,
  addOneVote,
}): ReactElement => {
  const { mostAndLeastVoted, optionsPercentage } = getVotesResults(
    group.options,
  );
  const firstIndex = 0;
  const singleOption = 1;
  const alphabet: string[] = 'abcdefghijklmnopqrstuvwxyz'.split('');

  const onOptionClick = (e: MouseEvent<HTMLButtonElement>): void => {
    addOneVote(e.currentTarget.id, group.id);
  };

  const optionClasses = classNames(styles['image-poll-view'], {
    'grid-cols-1': group.media.length !== firstIndex,
    'md:grid-cols-2 grid-cols-1': group.media.length === firstIndex,
  });
  return (
    <div className={optionClasses}>
      {group.media.length !== firstIndex ? (
        <SignleImagePollOption
          groupName={group.name}
          imageUrl={group.media[firstIndex].url}
          options={group.options}
          onOptionClick={onOptionClick}
        />
      ) : (
        group.options.map((option, index) => {
          const letter = alphabet[index];
          return (
            <ImagePollOption
              imageUrl={option.media[firstIndex].url}
              imgCaption={option.body}
              imgCaptionLetter={letter}
              optionId={option.id}
              leastVoted={option.vote_count === mostAndLeastVoted[singleOption]}
              mostVoted={option.vote_count === mostAndLeastVoted[firstIndex]}
              percentage={optionsPercentage[index]}
              isVoted={option.vote_count !== undefined}
              onOptionClick={onOptionClick}
            />
          );
        })
      )}
    </div>
  );
};

export default ImagePollGroup;
