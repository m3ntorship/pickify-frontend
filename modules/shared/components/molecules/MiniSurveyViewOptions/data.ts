export const optionsGroups = {
  groups: [
    {
      id: '03644270-7171-4147-b5a1-4233ff547f7a',
      name: 'Group Name',
      media: [],
      options: [
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
          body: 'ali tramsy',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 10,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
          body: ' ahmed essam',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 20,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
          body: ' omar gamal',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 30,
          voted: false,
        },
      ],
    },
  ],
};
