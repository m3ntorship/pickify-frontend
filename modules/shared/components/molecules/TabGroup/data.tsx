import React from 'react';
import { EPostType } from '../../../types/postFeed/EPostType';
import ImagePoll from '../../icons/imagePoll.svg';
import TextPoll from '../../icons/textPoll.svg';
import MiniSurvey from '../../icons/miniSurvey.svg';
import styles from '../../atoms/Tab/Tab.module.css';
import type { ITabGroup } from './ITabGroup';

export const tabGroupData = (): ITabGroup.ITabGroupData[] => {
  return [
    {
      id: '1',
      svg: <ImagePoll className={styles.icon} data-testid="image-poll-icon" />,
      value: 'Image Poll',
      postType: EPostType.ImagePoll,
    },
    {
      id: '2',
      svg: <TextPoll className={styles.icon} />,
      value: 'Text Poll',
      postType: EPostType.TextPoll,
    },
    {
      id: '3',
      svg: <MiniSurvey className={styles.icon} />,
      value: 'Mini Survey',
      postType: EPostType.MiniSurvey,
    },
  ];
};
