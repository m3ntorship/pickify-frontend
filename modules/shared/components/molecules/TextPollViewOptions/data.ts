export const apiDummyData = {
  caption: 'nalyzing Delaware Frozen',
  options_groups: {
    groups: [
      {
        id: '03644270-7171-4147-b5a1-4233ff547f7a',
        name: 'Group Name',
        options: [
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
            body: 'ali tramsi',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
            body: ' ahmed essam',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
            body: ' omar gamal',
          },
        ],
      },
    ],
  },
};
