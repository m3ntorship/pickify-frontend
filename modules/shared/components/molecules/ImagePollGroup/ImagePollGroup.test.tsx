import * as React from 'react';
import { render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import type { TargetElement } from '@testing-library/user-event';
import ImagePollGroup from './ImagePollGroup';
import { apiDummyData } from './data';

describe('OptionGroup component', () => {
  it('should render () OptionGroup when we apply ()', () => {
    const tree = renderer
      .create(<ImagePollGroup group={apiDummyData.group} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render 3 Options when we apply options 3 length', () => {
    const numberOfOptions = 3;

    render(<ImagePollGroup group={apiDummyData.group} />);

    const option: TargetElement[] = screen.getAllByTestId('image-poll-option');

    expect(option).toHaveLength(numberOfOptions);
  });
});
