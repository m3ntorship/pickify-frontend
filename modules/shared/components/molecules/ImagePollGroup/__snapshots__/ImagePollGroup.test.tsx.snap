// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionGroup component should render () OptionGroup when we apply () 1`] = `
<div
  className="image-poll-view md:grid-cols-2 grid-cols-1"
>
  <div
    className="relative w-auto"
    data-testid="image-poll-option"
  >
    <div
      className="image-view-wrapper"
    >
      <div
        className="bg-image-box"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "margin": 0,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "paddingTop": "100%",
              }
            }
          />
          <img
            alt="option"
            className="bg-image"
            decoding="async"
            id="https://placeimg.com/640/480/any"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "visibility": "hidden",
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="option"
          className="image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "contain",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="absolute bottom-4 right-4"
    >
      <div
        className="flex justify-end"
      >
        <button
          className="love group"
          id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white"
            data-file-name="SvgFilledHeart"
          />
        </button>
      </div>
    </div>
    <div
      className="absolute bottom-4 left-4"
    >
      <div
        className="captionContent"
      >
        <span
          className=""
        >
          a
        </span>
        <div
          className="captionDivider "
        />
        <span
          className="image-caption"
          dir="auto"
          title="ali tramsy"
        >
          ali tramsy
        </span>
      </div>
    </div>
  </div>
  <div
    className="relative w-auto"
    data-testid="image-poll-option"
  >
    <div
      className="image-view-wrapper"
    >
      <div
        className="bg-image-box"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "margin": 0,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "paddingTop": "100%",
              }
            }
          />
          <img
            alt="option"
            className="bg-image"
            decoding="async"
            id="https://placeimg.com/640/480/any"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "visibility": "hidden",
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="option"
          className="image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "contain",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="absolute bottom-4 right-4"
    >
      <div
        className="flex justify-end"
      >
        <button
          className="love group"
          id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white"
            data-file-name="SvgFilledHeart"
          />
        </button>
      </div>
    </div>
    <div
      className="absolute bottom-4 left-4"
    >
      <div
        className="captionContent"
      >
        <span
          className=""
        >
          b
        </span>
        <div
          className="captionDivider "
        />
        <span
          className="image-caption"
          dir="auto"
          title=" ahmed essam"
        >
           ahmed essam
        </span>
      </div>
    </div>
  </div>
  <div
    className="relative w-auto"
    data-testid="image-poll-option"
  >
    <div
      className="image-view-wrapper"
    >
      <div
        className="bg-image-box"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "margin": 0,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "paddingTop": "100%",
              }
            }
          />
          <img
            alt="option"
            className="bg-image"
            decoding="async"
            id="https://placeimg.com/640/480/any"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "visibility": "hidden",
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="option"
          className="image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "contain",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="absolute bottom-4 right-4"
    >
      <div
        className="flex justify-end"
      >
        <button
          className="love group"
          id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white"
            data-file-name="SvgFilledHeart"
          />
        </button>
      </div>
    </div>
    <div
      className="absolute bottom-4 left-4"
    >
      <div
        className="captionContent"
      >
        <span
          className=""
        >
          c
        </span>
        <div
          className="captionDivider "
        />
        <span
          className="image-caption"
          dir="auto"
          title=" omar gamal"
        >
           omar gamal
        </span>
      </div>
    </div>
  </div>
</div>
`;
